api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -164,6 +164,12 @@ message HttpConnectionManager {
   // timeout, although per-route idle timeout overrides will continue to apply.
   google.protobuf.Duration stream_idle_timeout = 24 [(gogoproto.stdduration) = true];
 
+  // A timeout for idle requests managed by the connection manager. The timer is activated when the
+  // request initiates, and is disarmed when the last byte of the request is sent upstream, OR when
+  // the response is encoded. If not specified or set to 0, this timeout is disabled. Retries
+  // are not currently supported.
+  google.protobuf.Duration request_timeout = 27 [(gogoproto.stdduration) = true];
```

Comment:
I had discussed this with @alyssawilk a while ago, but IMO with this we should consider deprecating the timer that is part of the buffer filter, as that is now redundant. Can we track that in a different issue?